table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -55,153 +97,537 @@ public DbConfig(@JsonProperty("bypass") @Nullable BypassEntry bypassEntry) {
-      Map<String, Set<String>> map = mEntries.stream().collect(
-          Collectors.toMap(BypassTableEntry::getTable, BypassTableEntry::getPartitions));
-      return new UdbBypassSpec(map);
+    public IgnoreTablesSpec(@Nullable IncludeExcludeList<NamePatternEntry> list) {
+      super(list);
     }
+  }
+
+  /**
+   * The "tables" object: {"tables": ... }.
+   * Base class for BypassTablesSpec and IgnoreTablesSpec.
+   * @param <T> the type of entry contained
+   */
+  public static  class TablesEntry<T extends NamePatternEntry> {
+    private final IncludeExcludeList<T> mList;
 
     /**
-     * @return tables bypassed
+     * @param list list of table entries
      */
-    public Set<String> getBypassedTables() {
-      return mEntries.stream().map(BypassTableEntry::getTable).collect(Collectors.toSet());
+    public TablesEntry(@Nullable IncludeExcludeList<T> list) {
+      mList = list == null ? new IncludeExcludeList<>(Collections.emptySet()) : list;
     }
 
     /**
-     * @return {@link BypassTableEntry}s
+     * Returns the list of tables.
+     * @return list of tables
      */
-    public Set<BypassTableEntry> getBypassTableEntries() {
-      return mEntries;
+    public IncludeExcludeList<T> getList() {
+      return mList;
     }
   }
 
   /**
-   * Table to partitions mapping.
+   * Deserialize to an IncludeExcludeList of given inner type T.
+   * When the contained object is an array, an include list is implied.
+   * @param <T> the contained inner entry type
    */
-  @JsonDeserialize(using = BypassTableEntryDeserializer.class)
-  public static class BypassTableEntry {
-    private final String mTableName;
-    private final Set<String> mPartitions;
+  static class TablesEntryDeserializer<T extends NamePatternEntry> {
+    IncludeExcludeList<T> deserializeToList(
+        Class<T> type, JsonParser jp, DeserializationContext cxt)
+        throws IOException, JsonProcessingException {
+      ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+      JsonNode node = mapper.readTree(jp);
+      if (node == null) {
+        return null;
+      }
+      if (!node.hasNonNull("tables")) {
+        throw new JsonParseException(mapper.treeAsTokens(node), "missing field `tables`");
+      }
+      node = node.get("tables");
+      if (node.isArray()) {
+        // in case of an array, an included list is implied
+        Set<T> entries = mapper.convertValue(
+            node,
+            mapper.getTypeFactory().constructCollectionType(Set.class, type)
+        );
+        return new IncludeExcludeList<>(entries);
+      }
+      if (node.isObject()) {
+        // otherwise, deserialize as an IncludeExcludeList object
+        return mapper.convertValue(
+            node,
+            mapper.getTypeFactory().constructParametricType(IncludeExcludeList.class, type)
+        );
+      }
+      throw new JsonParseException(mapper.treeAsTokens(node),
+          "invalid syntax, expecting array or object");
+    }
```

Comment:
The object has a required field `tables`, and it accept either a list of (implicitly) included tables, or an explicit inclusion and exclusion specification object.